go/arrow/array/table_test.go

```diff
@@ -157,7 +157,13 @@ func TestChunkedInvalid(t *testing.T) {
 		if e == nil {
 			t.Fatalf("expected a panic")
 		}
-		if got, want := e.(string), "arrow/array: mismatch data type float64 vs int32"; got != want {
+		errStr, ok := e.(string)
+		if !ok {
+			err := e.(error)
+			errStr = err.Error()
+		}
+
+		if got, want := errStr, fmt.Sprintf("%s: arrow/array: mismatch data type float64 vs int32", arrow.ErrInvalid); got != want {
```

Comment:
since we're expecting an error, this should probably be:

```go
err, ok := e.(error)
if !ok {
    t.Fatalf("expected an error")
}

if !errors.Is(err, arrow.ErrInvalid) {
    t.Fatalf("should be an ErrInvalid")
}

if got, want := err.Error(), fmt.Sprintf(........); got != want { ...
```